src/domain/usecases/SendNotificationDataValuesUseCase.ts

```diff
@@ -0,0 +1,347 @@
+import _ from "lodash";
+import jsonfile from "jsonfile";
+import log from "utils/log";
+import { DataValuesRepository } from "domain/repositories/DataValuesRepository";
+import { DataSetsRepository } from "domain/repositories/DataSetsRepository";
+import { OrgUnitRepository } from "domain/repositories/OrgUnitRepository";
+import { DataSet } from "domain/entities/DataSet";
+import { OrgUnit } from "domain/entities/OrgUnit";
+import { DataSetExecutionRepository } from "domain/repositories/DataSetExecutionRepository";
+import { promiseMap } from "data/dhis2-utils";
+import { UserRepository } from "domain/repositories/UserRepository";
+import { NotificationsRepository } from "domain/repositories/NotificationsRepository";
+import { DataValue } from "domain/entities/DataValue";
+import { User } from "domain/entities/User";
+import { Path } from "domain/entities/Base";
+import { Async } from "domain/entities/Async";
+import { Settings, SettingsRepository } from "domain/repositories/SettingsRepository";
+
+type EmailTemplate = {
+    subject: string;
+    body: string;
+};
+
+export type NotificationDetail = {
+    lastDateSentEmail: string;
+    lastUpdated: string;
+};
+
+export type NotificationStore = Record<string, NotificationDetail>;
```

Comment:
Not sure we need to store the dataValue we have notified. As I see it, we get the dataValues from the last notification marker, get the ones which match the time window, and update the marker.

Reply:
I don't get it. Here `lastUpdated` is being used to save the date of the last dataValue notified. should I save that value in a different format or attribute?